### ISSUE TITLE
Unexpected crash with exit code 1 when creating the PipelineLayout object.

### ISSUE THREAD

**Issue by Limeth** [vulkano-rs/vulkano#750]
[Working code](https://github.com/Limeth/vulkanscratch/tree/22d397aa4f09099c8fe1ecae842b260c97d82cdc) produces the following output, which is correct and expected:
https://hastebin.com/lebivamugo

While when I [uncomment a function](https://github.com/Limeth/vulkanscratch/commit/e4b51e33c9ec7c48ed81b26da48355464af7d668), the program hangs on the last line of output and after about a minute unexpectedly exits with exit code `1`. Output log:
https://hastebin.com/cuxececayi

The program hangs after printing out
`ObjectTracker information: OBJ[0x1e] : CREATE PipelineLayout object 0x15`
and exits before getting to printing out
`ObjectTracker information: OBJ[0x1f] : CREATE Pipeline object 0x16`

**Comment by tomaka** [comment]
Did you try enabling the validation layers, in case it reports something?

The most probable causes are a hardware limit that is not respected (vulkano doesn't check that yet), or a driver bug.

**Comment by Limeth** [comment]
Yes, I am enabling the validation layers, it just crashes without reporting anything. https://github.com/Limeth/vulkanscratch/blob/master/src/main.rs#L125

**Comment by tomaka** [comment]
For the record I tried running your code and got a stack overflow inside `std::ptr::write` when writing the data to upload to the immutable buffer.

**Comment by Limeth** [comment]
That's odd, I don't know what to make of that. Another hardware limitation, I guess.

**Comment by Limeth** [comment]
Does it happen right after `ObjectTracker information: OBJ[0x1e] : CREATE PipelineLayout object 0x15`, or do you think it's another issue?

**Comment by tomaka** [comment]
Maybe it's a legitimate stack overflow because there's too much data on the stack?

**Comment by tomaka** [comment]
If I call `ImmutableBuffer::uninitialized` instead of `from_data`, the stack overflow goes away and I get the same problem as you.

**Comment by tomaka** [comment]
Actually after letting it run I'm getting a different output:

> ObjectTracker information: OBJ[0x3] : CREATE Device object 0x269847a0020
> ObjectTracker information: OBJ[0x4] : CREATE VK_DEBUG_REPORT_OBJECT_TYPE_QUEUE_EXT object 0x26983f2f150
> ObjectTracker information: OBJ[0x5] : CREATE Buffer object 0x1
> ObjectTracker information: OBJ[0x6] : CREATE DeviceMemory object 0x2
> ObjectTracker information: OBJ[0x7] : CREATE Buffer object 0x3
> ObjectTracker information: OBJ[0x8] : CREATE Buffer object 0x4
> ObjectTracker information: OBJ[0x9] : CREATE Buffer object 0x5
> ObjectTracker information: OBJ[0xa] : CREATE Buffer object 0x6
> ObjectTracker information: OBJ[0xb] : CREATE DeviceMemory object 0x7
> ObjectTracker information: OBJ[0xc] : CREATE Buffer object 0x8
> ObjectTracker information: OBJ[0xd] : CREATE CommandPool object 0x9
> ObjectTracker information: OBJ[0xe] : CREATE VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_BUFFER_EXT object 0x26983e51db0
> ObjectTracker information: OBJ_STAT Destroy CommandBuffer obj 0x26983e51db0 (11 total objs remain & 0 CommandBuffer objs).
> ObjectTracker information: OBJ_STAT Destroy CommandPool obj 0x9 (10 total objs remain & 0 CommandPool objs).
> ObjectTracker information: OBJ_STAT Destroy Buffer obj 0x6 (9 total objs remain & 5 Buffer objs).
> ObjectTracker information: OBJ[0xf] : CREATE Buffer object 0xa
> ObjectTracker information: OBJ[0x10] : CREATE ShaderModule object 0xb
> ObjectTracker information: OBJ[0x11] : CREATE DescriptorSetLayout object 0xc
> ObjectTracker information: OBJ[0x12] : CREATE PipelineLayout object 0xd
> Driver error: Compute info
> ------------
> (0) : fatal error C9999: *** exception during compilation ***
> 
> ParameterValidation warning: vkCreateComputePipelines: returned VK_ERROR_INVALID_SHADER_NV, indicating that one or more shaders failed to compile or link
> thread 'main' panicked at 'Vulkan function returned VK_ERROR_INVALID_SHADER_NV', C:\Users\Pierre\.cargo\git\checkouts\vulkano-cb672043253a6e8d\c40954c\vulkano\src\lib.rs:23
> 3:39
> note: Run with `RUST_BACKTRACE=1` for a backtrace.
> ObjectTracker information: OBJ_STAT Destroy PipelineLayout obj 0xd (12 total objs remain & 0 PipelineLayout objs).
> ObjectTracker information: OBJ_STAT Destroy DescriptorSetLayout obj 0xc (11 total objs remain & 0 DescriptorSetLayout objs).
> ObjectTracker information: OBJ_STAT Destroy ShaderModule obj 0xb (10 total objs remain & 0 ShaderModule objs).
> ObjectTracker information: OBJ_STAT Destroy Buffer obj 0xa (9 total objs remain & 5 Buffer objs).
> ObjectTracker information: OBJ_STAT Destroy Buffer obj 0x8 (8 total objs remain & 4 Buffer objs).
> ObjectTracker information: OBJ_STAT Destroy Buffer obj 0x5 (7 total objs remain & 3 Buffer objs).
> ObjectTracker information: OBJ_STAT Destroy Buffer obj 0x4 (6 total objs remain & 2 Buffer objs).
> ObjectTracker information: OBJ_STAT Destroy Buffer obj 0x3 (5 total objs remain & 1 Buffer objs).
> ObjectTracker information: OBJ_STAT Destroy Buffer obj 0x1 (4 total objs remain & 0 Buffer objs).
> ObjectTracker information: OBJ_STAT Destroy DeviceMemory obj 0x7 (3 total objs remain & 1 DeviceMemory objs).
> ObjectTracker information: OBJ_STAT Destroy DeviceMemory obj 0x2 (2 total objs remain & 0 DeviceMemory objs).
> ObjectTracker information: OBJ_STAT Destroy Device obj 0x269847a0020 (1 total objs remain & 0 Device objs).
> ObjectTracker information: OBJ_STAT Destroy Queue obj 0x26983f2f150 (0 total objs remain & 0 Queue objs).
> loader debug: Unloading layer library C:\WINDOWS\system32\.\nvoglv64.dll
> loader debug: Unloading layer library D:\Logiciels\VulkanSDK\1.0.54.0\Bin\.\VkLayer_threading.dll
> loader debug: Unloading layer library D:\Logiciels\VulkanSDK\1.0.54.0\Bin\.\VkLayer_parameter_validation.dll
> loader debug: Unloading layer library D:\Logiciels\VulkanSDK\1.0.54.0\Bin\.\VkLayer_object_tracker.dll
> loader debug: Unloading layer library D:\Logiciels\VulkanSDK\1.0.54.0\Bin\.\VkLayer_core_validation.dll
> loader debug: Unloading layer library D:\Logiciels\VulkanSDK\1.0.54.0\Bin\.\VkLayer_unique_objects.dll
> error: process didn't exit successfully: `target\debug\vulkanscratch.exe` (exit code: 101)

The interesting part is unfortunately not helpful:

> (0) : fatal error C9999: *** exception during compilation ***

I guess your best bet is to report this to nVidia: https://devtalk.nvidia.com/default/board/166/vulkan/

**Comment by Limeth** [comment]
Will do, thank you.
